4/bin/entrypoint.sh

```diff
@@ -1,7 +1,20 @@
 #!/bin/bash
 
-set -e
-set -o pipefail
+set -euo pipefail
+
+declare -a args
+
+add_env_var_as_env_prop() {
+  if [ ! -z "$1" ]; then
+    args+=("-D$2=$1")
+  fi
+}
+
+add_env_var_as_env_prop "${SONAR_HOST_URL:-}" "sonar.host.url"
+add_env_var_as_env_prop "${SONAR_LOGIN:-}" "sonar.login"
+add_env_var_as_env_prop "${SONAR_PASSWORD:-}" "sonar.password"
+add_env_var_as_env_prop "${SONAR_USER_HOME:-}" "sonar.userHome"
```

Comment:
I would add also `SONAR_SOURCE` env variable which maps to sonar.source to allow users define root source in other place then `/usr/src`.

We were trying to test this image on gitlab ci and it looks like CI is mounting volume into different directory than `/usr/src` which ends up with error.

Reply:
I'm not sure I see the benefit of providing an env var to define `sonar.sources`.

1. because we run the scanner in the container, `sonar.sources` can only point to a location in the container.
2. to use the scanner's container, you need to bind a local directory (or a volume) to a location inside the container. That's what the option `-v "/path/to/project:/usr/src"` of the `docker run` does.

If one specifies a different `sonar.sources`, that will basically allow users to change the second part of the option, eg.: `-v "/path/to/project:/usr/src2"` -e SONARSOURCES=/usr/src2

Why would anyone do that?

Reply:
or better would be `sonar.projectBaseDir` as sonar.sources points only to sources omitting tests or some other way to set project root (but seems out of scope of this issue)

Reply:
As discussed if you run this image inside some CI there is a chance that you don't control where sources will be mounted (for example Gitlab CI)

Reply:
> Why would anyone do that?

to answer my own question after discussing with @jacek-poreda-sonarsource , one might want to specify the path to the project directory inside the container when one does not control the mount point of the directory (eg. on Gitlab CI)